sdks/python/apache_beam/io/avroio.py

```diff
@@ -137,6 +139,26 @@ def offset(self):
       return self._offset
 
   def read_records(self, file_name, range_tracker):
+    current_block_offset = -1
+    current_block_size = -1
+
+    def _callback(stop_position):
+      if range_tracker.done():
+        return 0
+
+      # Locking progressing to next block  to provide a consistent result
+      with self._progress_lock:
+        if (current_block_offset + current_block_size >=
+            range_tracker.stop_position()):
+          # This block ends after the last byte that belongs to the position
+          # range of the source. Hence this has to be the last block that
+          # belongs to the position range.
+          return 1
+        return filebasedsource.FileBasedSource.remaining_split_points_helper(
```

Comment:
Would this return anything other than `iobase.RangeTracker.SPLIT_POINTS_UNKNOWN`. It seems like the first two conditions are already checked here.